File: src/my_py_pkg/my_py_pkg/robo_news_rec.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class RoboRec(Node):
    def __init__(self):
        super().__init__("robo_news_rec")
        self.subscriber_ = self.create_subscription(
            String, "robot_news", self.newsRecCallback, 10
        )
        self.get_logger().info("News Receiver Started")

    def newsRecCallback(self, msg: String):
        self.get_logger().info(msg.data)


def main(args=None):
    rclpy.init(args=args)
    node = RoboRec()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/robo_news_rec.cpp
#include "example_interfaces/msg/string.hpp"
#include "rclcpp/rclcpp.hpp"
class RoboNewsRec : public rclcpp::Node {
public:
  RoboNewsRec() : rclcpp::Node("robot_news_rec_cpp") {
    subscriber_ = this->create_subscription<example_interfaces::msg::String>(
        "robot_news", 10,
        std::bind(&RoboNewsRec::subscriptionCallback, this,
                  std::placeholders::_1));
    RCLCPP_INFO(get_logger(), "reciever started");
  }

private:
  rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;
  void subscriptionCallback(example_interfaces::msg::String::SharedPtr msg) {
    RCLCPP_INFO(get_logger(), "%s", msg->data.c_str());
  }
};

int main(int argc, char **argv) {
  rclcpp::init(argc, argv);
  auto node = std::make_shared<RoboNewsRec>();
  rclcpp::spin(node);
  rclcpp::shutdown();
}
